### ISSUE TITLE
(Anisotropic Polynomial Expansion) orth_ttr: order as vector

### ISSUE THREAD

**Issue by debusschercharles** [jonathf/chaospy#236]
Hi Jonathan,

I am trying to create to create a polynomial expansion with different weights for the different dimensions, i.e. an anisotropic polynomial expansion.

The issue arises when an input order-vector of more than 2 dimensions is given to the truncation scheme. The code bellow illustrates the issue with a basic trivariate expansion.

```
>>> distribution = chaospy.J(chaospy.Normal(0, 1), chaospy.Normal(0, 1), chaospy.Normal(0, 1))
>>> print(chaospy.orth_ttr([2, 3, 4], distribution))
```

The following error is generated:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-10-07709c9d32ef> in <module>
      1 import chaospy
      2 distribution = chaospy.J(chaospy.Normal(0, 1), chaospy.Normal(0, 1), chaospy.Normal(0, 1))
----> 3 print(chaospy.orth_ttr([2, 3, 4], distribution))

/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/chaospy/orthogonal/three_terms_recursion.py in orth_ttr(order, dist, normed, sort, retall, cross_truncation, **kws)
    109 
    110     order = numpy.array(order)
--> 111     indices = numpoly.bindex(start=0, stop=order+1, dimensions=len(dist),
    112                              ordering=sort, cross_truncation=cross_truncation)
    113     if len(dist) > 1:

/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpoly/poly_function/monomial/bindex.py in bindex(start, stop, dimensions, ordering, cross_truncation)
     48 
     49     cross_truncation = cross_truncation*numpy.ones(2)
---> 50     indices = _bindex(start, stop, dimensions, cross_truncation)
     51 
     52     if "G" in ordering:

/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpoly/poly_function/monomial/bindex.py in _bindex(start, stop, dimensions, cross_truncation)
     86 
     87         # Truncate at each step to keep memory usage low
---> 88         indices = indices[cross_truncate(indices, stop-1, cross_truncation[1])]
     89 
     90     # Complete the truncation scheme

/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpoly/poly_function/monomial/cross_truncation.py in cross_truncate(indices, bound, norm)
     47         out = numpy.max(indices/bound, axis=-1) <= 1
     48     else:
---> 49         out = numpy.sum((indices/bound)**norm, axis=-1)**(1./norm) <= 1
     50     if not numpy.any(bound < 0):
     51         out[numpy.all(indices == 0, axis=-1)] = True

ValueError: operands could not be broadcast together with shapes (25,2) (3,) 
```

It would be very useful to me if it would be possible to generate anisotropic polynomial expansions with more than 2 dimensions.

I am using Python 3.8.2 with ChaosPy 3.2.10.

Thanks!

__________________________________________________________________________________________________________
Side note:
Moreover, when running the examples shown in the documentation on p.69 in the Feb, 17 2020 edition, I get different results. It is my intuition that the examples given in the documentation are incorrect. 

My run:
```
>>> distribution = chaospy.J(chaospy.Normal(0, 1), chaospy.Normal(0, 1))
>>> print(chaospy.orth_ttr([0, 2], distribution))
[1.0]
>>> print(chaospy.orth_ttr([1, 2], distribution))
[1.0 q1 q0 -1.0+q1**2]
>>> print(chaospy.orth_ttr([2, 2], distribution))
[1.0 q1 q0 -1.0+q1**2 q0*q1 -1.0+q0**2]
```

Documentation:
```

>>> print(chaospy.orth_ttr([0, 2], distribution))
[1.0, q0, q0^2-1.0]
>>> print(chaospy.orth_ttr([1, 2], distribution))
[1.0, q1, q0, q0q1, q0^2-1.0]
>>> print(chaospy.orth_ttr([2, 2], distribution))
[1.0, q1, q0, q1^2-1.0, q0q1, q0^2-1.0]
```

**Comment by jonathf** [comment]
That is a bug. I have tried to fix it in v3.2.11 now.

Thanks for pointing out the docs-issue. They should have be covered by doctesting, but looks like pylint wasn't pointing to this document. It should be fixed now.

I am working on migrating the tutorial stuff to a collection of ipython notebooks. The anisotropicity/cross-truncation stuff is now moved here:
https://github.com/jonathf/chaospy/blob/master/tutorial/polynomial/truncation.ipynb
